internal/pkg/utils/testproject/project.go

```diff
@@ -86,40 +109,62 @@ func GetTestProject(t *testing.T, envs *env.Map) *Project {
 	// Check token/project ID
 	initWg := &sync.WaitGroup{}
 	initWg.Add(1)
+	errs := make(chan error, 1)
+	wgDone := make(chan bool)
 	go func() {
 		defer initWg.Done()
 		if token, err := storageapi.VerifyTokenRequest(p.Project.StorageAPIToken()).Send(p.ctx, p.storageApiClient); err != nil {
-			assert.FailNow(p.t, fmt.Sprintf("invalid token for project %d: %s", p.ID(), err))
+			errs <- fmt.Errorf("invalid token for project %d: %s", p.ID(), err)
```

Comment:
Ak chapem spravne, tak tato inicializacia bezi paralelne a chces vratit prvu chybu co nastane.

Nato existuje standardny mechanizmus `errgrp`:
https://pkg.go.dev/golang.org/x/sync/errgroup

----- 

Inak, pri prvej chybe zavolas `close(errs)`, lenze nic ti negarantuje, ze o `1us` nenastane dalsia chyba - a to by nastal panic, ze zapisujes do uzatvoreneho kanala

----

Ina alternativa je pouzit `multierror` a zapisovat z goroutine do spolocnej premennej `errs` (multierror).
Multi error obsahuje zamky, takze je to bezpecne.
Takto by si zozbieral a vratil vsetky chyby.

Reply:
No mně šlo jen o to, že když jsem přišel o možnost zavolat `assert.Fail()`, tak tu chybu potřebuju zpropagovat z go rutiny ven. A tohle jsem našel hned v několika článcích, tak jsem myslel že se to tak používá. Aha, ale vypadá to že s tou `multierror` to bude rozhodně příjemnější. 🙂

Reply:
> A tohle jsem našel hned v několika článcích, tak jsem myslel že se to tak používá.

Nie je to v principe zle riesenie, ... ak vies nejak garantovat, ze nebudes zapisovat do zatvoreneho kanala, co mozno v tych clankoch bolo nejak garantovane.

Reply:
Taky fixnuto: https://github.com/keboola/keboola-as-code/pull/752/commits/3862d3786deb013fe0a65308b45258d5aca5d01d